lcmgen/lcm-gen.1

```diff
@@ -14,124 +14,161 @@ unmarshalling messages of the specified types.
 
 Currently, \fBlcm-gen\fR is capable of generating language bindings for C, C++,
 Java, Python, Lua, and C#.
-
-.SH GENERAL OPTIONS
 .TP
-.B \-h
-Prints some help text and exits.
+\fB\-h\fR | \fB\-\-help\fR
+[ true ]                            Show this help
 .TP
-.B \-t, \-\-tokenize
-Show tokenization.
+\fB\-t\fR | \fB\-\-tokenize\fR
+[ false ]                           Show tokenization
 .TP
-.B \-d, \-\-debug
-Show parsed files.
+\fB\-d\fR | \fB\-\-debug\fR
+[ false ]                           Show parsed file
 .TP
-.B \-\-lazy
-Generate output file only if .lcm is newer than output file (or if output file
-does not already exist).
+\fB\-\-lazy\fR
+[ false ]                           Generate output file only if .lcm is newer
 .TP
-.B \-\-package\-prefix \fIPFX\fR
-Add package name \fIPFX\fR as a prefix to the declared package.
+\fB\-\-use\-quotes\-for\-includes\fR
+[ false ]                           Use quotes instead of angular brackets for including header files
 .TP
-.B \-\-version
-Shows version information and exits.
-
-.SH C OPTIONS
+\fB\-\-package\-prefix\fR
+[  ]                                Add this package name as a prefix to the declared package
 .TP
-.B \-c, \-\-c
-Emit C code.
+\fB\-\-version\fR
+[ false ]                           Show version information and exit
+.IP
+**** C options ****
```

Comment:
Nitpick: The previous version is more inline with what I'd expect in a manpage.  Meaning, **C OPTIONS** rather than an indented `**** C OPTIONS ****`.

Reply:
The hard-coded manpage used the troff macro `.SH C Options` to denote those as new sections. The same options apply here as in [this thread](https://github.com/lcm-proj/lcm/pull/538#discussion_r1829454188), so I imagine whatever we do there will also happen here.

Reply:
This is a bit trickier.  This is my suggestion here, though, I don't love it.

```diff
diff --git a/lcmgen/getopt.c b/lcmgen/getopt.c
index 176ed77..c0e6cb9 100644
--- a/lcmgen/getopt.c
+++ b/lcmgen/getopt.c
@@ -291,7 +291,7 @@ void getopt_do_usage(getopt_t *gopt)
             if (goo->help == NULL || strlen(goo->help) == 0)
                 printf("\n");
             else
-                printf("\n%*s%s\n\n", leftmargin, "", goo->help);
+                printf("\n%s\n\n", goo->help);
             continue;
         }
 
diff --git a/lcmgen/main.c b/lcmgen/main.c
index b579b73..0416944 100644
--- a/lcmgen/main.c
+++ b/lcmgen/main.c
@@ -53,31 +53,31 @@ int main(int argc, char *argv[])
         "Add this package name as a prefix to the declared package");
     getopt_add_bool(gopt,   0, "version",  0, "Show version information and exit");
 
-    getopt_add_spacer(gopt, "**** C options ****");
+    getopt_add_spacer(gopt, "C OPTIONS");
     getopt_add_bool(gopt, 'c', "c",        0, "Emit C code");
     setup_c_options(gopt);
 
-    getopt_add_spacer(gopt, "**** C++ options ****");
+    getopt_add_spacer(gopt, "C++ OPTIONS");
     getopt_add_bool(gopt, 'x', "cpp",      0, "Emit C++ code");
     setup_cpp_options(gopt);
 
-    getopt_add_spacer(gopt, "**** Java options ****");
+    getopt_add_spacer(gopt, "Java OPTIONS");
     getopt_add_bool(gopt, 'j', "java",     0, "Emit Java code");
     setup_java_options(gopt);
 
-    getopt_add_spacer(gopt, "**** Python options ****");
+    getopt_add_spacer(gopt, "Python OPTIONS");
     getopt_add_bool(gopt, 'p', "python",   0, "Emit Python code");
     setup_python_options(gopt);
 
-    getopt_add_spacer(gopt, "**** Lua options ****");
+    getopt_add_spacer(gopt, "Lua OPTIONS");
     getopt_add_bool(gopt, 'l', "lua",      0, "Emit Lua code");
     setup_lua_options(gopt);
 
-    getopt_add_spacer(gopt, "**** C#.NET options ****");
+    getopt_add_spacer(gopt, "C#.NET OPTIONS");
     getopt_add_bool(gopt, 0, "csharp",     0, "Emit C#.NET code");
     setup_csharp_options(gopt);
 
-    getopt_add_spacer(gopt, "**** Go options ****");
+    getopt_add_spacer(gopt, "Go OPTIONS");
     getopt_add_bool(gopt, 'g', "go",       0, "Emit Go code");
     setup_go_options(gopt);
     // clang-format on
```

`--help` output for the `C` options,

```
C OPTIONS

  -c | --c                        [ false ]                           Emit C code
       --c-cpath                  [ . ]                               Location for .c files
       --c-hpath                  [ . ]                               Location for .h files
       --c-export-include         [  ]                                #include that provides the export symbol
       --c-export-symbol          [  ]                                ABI export decoration symbol
       --cinclude                 [  ]                                Generated #include lines reference this folder
       --c-no-pubsub              [ false ]                           Do not generate _publish and _subscribe functions
       --c-typeinfo               [ false ]                           Generate typeinfo functions for each type
```

and the `man` page,

```
       C OPTIONS

       -c | --c
              [ false ]                           Emit C code
```

Reply:
Added this change via 4bb4358dfd0a8ff1679daa019f8559f744dbf23e